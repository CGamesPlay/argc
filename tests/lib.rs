use insta::assert_snapshot;

macro_rules! assert_argc {
    (
        $source:expr,
        $args:expr
    ) => {
        let (stdout, stderr) = argc::run($source, $args).unwrap();
        let args = $args.join(" ");
        let stdout = stdout.unwrap_or_default();
        let stderr = stderr.unwrap_or_default();
        let output = format!(r###"RUN
{}

STDOUT
{}

STDERR
{}
"###, args, stdout, stderr);
        assert_snapshot!(output);
    };
}

#[test]
fn test_git() {
    assert_argc!(include_str!("git.sh"), &["git", "-h"]);
}

#[test]
fn test_git_add() {
    assert_argc!(include_str!("git.sh"), &["git", "add", "-h"]);
}

#[test]
fn test_git_remote() {
    assert_argc!(include_str!("git.sh"), &["git", "push", "-h"]);
}

#[test]
fn test_git_log() {
    assert_argc!(include_str!("git.sh"), &["git", "log", "-h"]);
}