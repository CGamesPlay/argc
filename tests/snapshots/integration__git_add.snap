---
source: tests/lib.rs
assertion_line: 32
expression: output

---
RUN
git add -h

STDOUT
exit 1

STDERR
git-add 
Add file contents to the index

USAGE:
    git add [OPTIONS] [pathspec]...

ARGS:
    <pathspec>...    Files to add content from.

OPTIONS:
    -h, --help       Print help information
    -n, --dry-run    Donâ€™t actually add the file


